### PR TITLE
sequencer: Provide instructions to --continue if a commit doesn't mer…

### PR DESCRIPTION
**Pull request by stephencmorton**
…ge cleanly

During a rebase or cherry-pick of multiple commits, the error message will now
correctly tell you to run <operation> --continue rather than just 'git commit'.

Signed-off-by: Stephen Morton stephen.morton@nokia.com

### CODE REVIEW

File: sequencer.c [hunk 1]
```diff
@@ -170,13 +170,16 @@ static void print_advice(int show_hint, struct replay_opts *opts)
...
-		if (opts->no_commit)
-			advise(_("after resolving the conflicts, mark the corrected paths\n"
-				 "with 'git add <paths>' or 'git rm <paths>'"));
-		else
-			advise(_("after resolving the conflicts, mark the corrected paths\n"
-				 "with 'git add <paths>' or 'git rm <paths>'\n"
-				 "and commit the result with 'git commit'"));
+		advise(_("after resolving the conflicts, mark the corrected paths\n"
+			"with 'git add <paths>' or 'git rm <paths>'"));
+		if (! opts->no_commit)
```
Comment (gitster) [inline sequencer.c]: In our codebase, opening { goes at the end of if (...) line that is before it.

File: sequencer.c [hunk 2]
```diff
@@ -154,7 +154,7 @@ static void free_message(struct commit *commit, struct commit_message *msg)
 	unuse_commit_buffer(commit, msg->message);
 }
 
-static void print_advice(int show_hint, struct replay_opts *opts)
+static void print_advice(int show_hint, struct replay_opts *opts, int last_commit)
```
Comment (gitster) [inline sequencer.c]: This is probably misnamed. You seem to be using it as a boolean that lets this function tell if this is the last commit, but it sounds as if you are referring to nth commit in a sequence and passing that number to this function.

Perhaps rename it "is_last" or something?

**Comment by dscho** [comment]
> Please send it to [git@vger.kernel.org](mailto:git@vger.kernel.org) as a submission for a formal review.

@stephencmorton are you still interested in this PR?

**Comment by stephencmorton** [comment]
> > Please send it to [git@vger.kernel.org](mailto:git@vger.kernel.org) as a submission for a formal review.
> 
> @stephencmorton are you still interested in this PR?

I suspect that git/git has already solved this in some other way.

If it has not... I _was_ interested in the PR, but my diffs kept getting rejected. All for reasonable and good reasons of ensuring quality code, to insist on Doing The Right Thing (TM), but I only have so much motivation to dive into the git code and learn what the right thing to do is. If somebody who knows the git code wants to pick this up and do it right, I'm all for it, but I have lost motivation, given the very small nature of this change.

**Comment by dscho** [comment]
@stephencmorton I understand.

There is currently a GSoC project going on that hopefully addresses this. If not, I am sure that @r1walz will be very willing to look into this.

**Comment by r1walz** [comment]
@stephencmorton: Yes, you are correct. git/git has already addressed this issue to some extent (like for rebase -i). I am currently working on improving consistency among these commands. I'm sure when I'm done with my project this issue will be fully resolved.

@dscho: Currently, I've not corrected all the advice messages since it requires a little more knowledge of the code base which I lack. I'll attend to this after all the main points of the proposal are checked.

**Pull request closed without merging**